Top performing students HTML

QueenWoofie opened the pull request:
I've added the HTML to the top performing students page. 

I haven't added the href's yet, but planned to do that next as I go through and confirm where they all lie. 
I also plan to add a little tailwind CSS next.

QueenWoofie committed:
added .html file to public folder as per QA's instructions

QueenWoofie committed:
Update top-performing-students

added breadcrumbs, heeading, body-text, search bar, table, pagination, dashboard button link as html.

QueenWoofie committed:
Merge branch 'develop' into top-performing-students

QueenWoofie committed:
Added Tailwind CSS to top-performing-students

I've added some general styles for desktop using tailwind CSS
- See pull request for further commentary

QueenWoofie commented:
Update 10/01/2025 - Added some tailwind CSS elements to the page, but only for desktop currently. 

What still needs to get done:

* Adding **Header**, **Footer** and **Sidebar**
* Adding an icon in the search bar input
* Adding responsiveness to adjust to mobile layout
* Fix fonts to apply the correct ones
* Add hover and other reaction effects to elements where this applies - such as buttons and links

* Double check that the HTML page is in the right place
> Originally it was said to be put in src/ but now they seem to have been moved.

* Double check CSS link in the HTML file
> I couldn't get the tailwind CSS to work at first, but eventually I managed to get it to work with output.css, but depends on what has been done in regards to the tailwind files after.

* Add links in the href / src elements that are currently set as "#", as the placement of them was unsure at the time as I was creating this.
* Adding javascript to add actual functionality to the elements of the page that require this.

sulenchy commented:
Pls, attach its ticket to the PR description @QueenWoofie

Tomatsuppe11211 reviewed (dismissed):
Change breadcrumb

Your code:
![image](https://github.com/user-attachments/assets/f30d8ada-cccf-4dda-83b0-2f852d693545)

figma design:
![image](https://github.com/user-attachments/assets/5d728762-fc97-42e1-8c71-caeee9de2eac)


With the searchbar there is a missing icon (searchglass)

All the other elements are included :)


When it comes to the tailwind CSS the code is now outdated because of the v4 update. This has to be changed here as well.  After that we can work on scaling (making this page responsive)

Tomatsuppe11211 committed:
Merge branch 'develop' into top-performing-students

Tomatsuppe11211 committed:
Updated top performing students

I have updated this file and branch to follow tailwind V4. At the same time, I have been working with scaling the pager. I have added another table for mobile version. There are some issues with changing font sizes to h1- and p-elements.

Tomatsuppe11211 committed:
Merge branch 'develop' into top-performing-students

Tomatsuppe11211 commented:
I have some issues here with changing font sizes on p-elements and h1-elements when sccaling to mobile version.

sulenchy commented:
> I have some issues here with changing font sizes on p-elements and h1-elements when sccaling to mobile version.

@Tomatsuppe11211 that is because font size for h1 and p tags have been set in the css/styles.css file. As it is, the breakpoint won't applies unless you adjust the styling in the file. There are two options: You can remove the h1 and p styling and update all h1 and p tags across the codebase using the tailwind classes or you remove just the font-size and fix only the font-size across the codebase. 

However, let's merge this PR like that and create an issue to fix it. I will create an issue to resolve it.

Tomatsuppe11211 commented:
> > I have some issues here with changing font sizes on p-elements and h1-elements when sccaling to mobile version.
> 
> @Tomatsuppe11211 that is because font size for h1 and p tags have been set in the css/styles.css file. As it is, the breakpoint won't applies unless you adjust the styling in the file. There are two options: You can remove the h1 and p styling and update all h1 and p tags across the codebase using the tailwind classes or you remove just the font-size and fix only the font-size across the codebase.
> 
> However, let's merge this PR like that and create an issue to fix it. I will create an issue to resolve it.
> 
> ok :)

Tomatsuppe11211 commented:
ok

sulenchy commented:
@Tomatsuppe11211 Here is the ticket https://github.com/NoroffFEU/edu-gate/issues/340

Tomatsuppe11211 commented:
nice! shall i do it now?

sulenchy commented:
> nice! shall i do it now?

No, you are not obligated to work on it. But you can if you wish

sulenchy reviewed (approved):
LGTM

kristinatonnessen committed:
Merge branch 'develop' into top-performing-students
